src/adapter/src/session/vars.rs

```diff
@@ -1428,6 +1451,41 @@ impl Value for [String] {
     }
 }
 
+impl Value for Vec<String> {
+    const TYPE_NAME: &'static str = "string list";
+
+    fn parse(s: &str) -> Result<Vec<String>, ()> {
+        match parse_set_variable_value(s) {
```

Comment:
I think we should save ourselves a non-trivial amount of duplicated effort for implementing parsing and formatting logic for various `Value` implementations if we just delegate that to the `Display` and `Parser` logic of the `SetVariableValue` structure that models the corresponding `Value` type.

Reply:
That seems reasonable. The coupling exist whether or not we make it explicit, in that we'll have to always keep the two implementations in sync. We might as well make it explicit.